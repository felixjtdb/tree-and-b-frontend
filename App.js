import React, { Component } from 'react'
import { StyleSheet, Text, View } from 'react-native'
import SwipeableViews from 'react-swipeable-views-native'

import Tree from './components/Tree'
import Forest from './components/Forest'
import TreeViewer from './components/TreeViewer'
import Axe from './components/Axe'

const ALL_TREES_URL = 'localhost:3000/trees'

export default class App extends Component {
  state = {
    testForest :
      [ { id : 1,
          name: 'Oak',
          imageURL: 'https://www.fast-growing-trees.com/images/P/Live-Oak-Tree-450w.jpg',
          description: "I'm a big ol tree",
          location: 123
        },
        { id : 2,
          name: 'BrOak',
          imageURL: "http://www.irishtreesociety.com/wp-content/uploads/irish-tree-society-feature.jpg",
          description: "me too!",
          location: 456
        }
      ],
      data: ''
  }

  componentDidMount = () => {
    fetch(ALL_TREES_URL, {
      method: 'GET'
    })
    .then((response) => response.json())
    .then((responseJSON) => {
      this.setState({
        data: responseJSON.body
      })
    })
    .catch(error => {
      this.setState({
        data: 'ERROR WHEN FETCHING TREES... THE FOREST IS ON FIRE!!'
      })
    })
  }

  render() {
    return (
      <TreeViewer trees={this.state.testForest} />
      // <SwipeableViews>
      //
      //     <Forest trees={this.state.testForest}/>
      // </SwipeableViews>

    );
  }
}

const styles = StyleSheet.create({
  titleText: {
    fontSize: 20,
    fontWeight: 'bold',
  },
  defaultSpacing: {
    top: 50
  }
});

// <TreeViewer imageURL={'http://www.irishtreesociety.com/wp-content/uploads/irish-tree-society-feature.jpg'}/>
