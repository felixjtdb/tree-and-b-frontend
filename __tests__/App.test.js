import React, { Component } from 'react';
import App from '../App.js'
import renderer from 'react-test-renderer';

test('renders correctly', () => {
  const app = renderer.create(<App />).toJSON();
  expect(app).toMatchSnapshot();
});
