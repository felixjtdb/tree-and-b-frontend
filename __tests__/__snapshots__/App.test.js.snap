// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<View
  style={
    Object {
      "alignItems": "center",
      "backgroundColor": "transparent",
      "flex": 1,
      "justifyContent": "center",
    }
  }
>
  <View
    onMoveShouldSetResponder={[Function]}
    onMoveShouldSetResponderCapture={[Function]}
    onResponderEnd={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderReject={[Function]}
    onResponderRelease={[Function]}
    onResponderStart={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    onStartShouldSetResponderCapture={[Function]}
    style={
      Object {
        "opacity": 1,
        "transform": Array [
          Object {
            "translateX": 0,
          },
          Object {
            "translateY": 0,
          },
          Object {
            "rotate": "0deg",
          },
          Object {
            "scale": 0.5,
          },
        ],
      }
    }
  >
    <View
      style={
        Object {
          "alignItems": "center",
          "backgroundColor": "#ecf0f1",
          "flex": 1,
          "justifyContent": "center",
          "paddingTop": 20,
        }
      }
    >
      <Image
        resizeMode="cover"
        source={
          Object {
            "uri": "https://www.fast-growing-trees.com/images/P/Live-Oak-Tree-450w.jpg",
          }
        }
        style={
          Object {
            "borderRadius": 150,
            "height": 150,
            "width": 150,
          }
        }
      />
      <Text
        style={
          Object {
            "color": "red",
          }
        }
      >
         
        Oak
         
      </Text>
      <Text
        style={Object {}}
      >
         
        I'm a big ol tree
         
      </Text>
      <Text
        style={Object {}}
      >
         
        123
         
      </Text>
    </View>
  </View>
  <View
    style={
      Object {
        "borderColor": "red",
        "borderRadius": 5,
        "borderWidth": 2,
        "bottom": 20,
        "left": 0,
        "opacity": 0,
        "padding": 20,
        "position": "absolute",
        "transform": Array [
          Object {
            "scale": 0,
          },
        ],
      }
    }
  >
    <Text
      style={
        Array [
          Object {
            "color": "red",
            "fontSize": 16,
          },
          undefined,
        ]
      }
    >
      Nope!
    </Text>
  </View>
  <View
    style={
      Object {
        "borderColor": "green",
        "borderRadius": 5,
        "borderWidth": 2,
        "bottom": 20,
        "opacity": 0,
        "padding": 20,
        "position": "absolute",
        "right": 0,
        "transform": Array [
          Object {
            "scale": 0.5,
          },
        ],
      }
    }
  >
    <Text
      style={
        Array [
          Object {
            "color": "green",
            "fontSize": 16,
          },
          undefined,
        ]
      }
    >
      Yup!
    </Text>
  </View>
</View>
`;
